block/block_cache.go

```diff
@@ -9,40 +9,50 @@ import (
 type Cache struct {
 	cache map[uint64]CachedBlock
 	sync.Mutex
-	lowestCachedBlockHeight  uint64
-	highestCachedBlockHeight uint64
 }
 
 func (m *Cache) AddBlockToCache(h uint64, b *types.Block, c *types.Commit) {
 	m.Lock()
 	defer m.Unlock()
 	m.cache[h] = CachedBlock{Block: b, Commit: c}
 
-	types.NumberOfAccumulatedBlocks.Set(float64(len(m.cache)))
-	if m.lowestCachedBlockHeight == 0 || h < m.lowestCachedBlockHeight {
-		m.lowestCachedBlockHeight = h
-		types.LowestPendingBlockHeight.Set(float64(h))
-	}
-	if h > m.highestCachedBlockHeight {
-		m.highestCachedBlockHeight = h
-		types.HighestReceivedBlockHeight.Set(float64(h))
-	}
+	types.BlockCacheSize.Set(float64(len(m.cache)))
+	types.LowestPendingBlockHeight.Set(m.GetLowestCachedBlockHeight())
+	types.HighestReceivedBlockHeight.Set(m.GetHighestCachedBlockHeight())
 }
 
 func (m *Cache) DeleteBlockFromCache(h uint64) {
 	m.Lock()
 	defer m.Unlock()
 	delete(m.cache, h)
 
-	types.NumberOfAccumulatedBlocks.Set(float64(len(m.cache)))
-	if h == m.lowestCachedBlockHeight {
-		m.lowestCachedBlockHeight = h + 1
-		types.LowestPendingBlockHeight.Set(float64(m.lowestCachedBlockHeight))
+	types.BlockCacheSize.Set(float64(len(m.cache)))
+	types.LowestPendingBlockHeight.Set(float64(m.GetLowestCachedBlockHeight()))
+	types.HighestReceivedBlockHeight.Set(float64(m.GetHighestCachedBlockHeight()))
+}
+
+func (m *Cache) GetLowestCachedBlockHeight() float64 {
+	m.Lock()
+	defer m.Unlock()
+	var lowest uint64
+	for h := range m.cache {
```

Comment:
why we iterate here vs just saving the lowest on first entrance and comparing each new block to it?